src/context/statistics.tsx

```diff
@@ -0,0 +1,72 @@
+import React, {
+  createContext,
+  FunctionComponent,
+  useState,
+  useCallback
+} from "react";
+export const FEATURES_KEY = "FEATURES";
+
+interface StatisticsContext {
+  totalCount: number | null;
+  currentTimestamp: number;
+  lastTransactionTime: number | null;
+  transactionHistory: { name: string; category: string; quantity: number }[];
+  setTotalCount: (totalCount: number | null) => void;
+  setCurrentTimestamp: (currentTimestamp: number) => void;
+  setLastTransactionTime: (lastTransactionTime: number) => void;
+  setTransactionHistory: (
+    transactionHistory: { name: string; category: string; quantity: number }[]
+  ) => void;
+  clearStatistics: () => void;
+}
+export const StatisticsContext = createContext<StatisticsContext>({
+  totalCount: null,
+  currentTimestamp: Date.now(),
+  lastTransactionTime: null,
+  transactionHistory: [],
+  setTotalCount: () => null,
+  setCurrentTimestamp: () => null,
+  setLastTransactionTime: () => null,
+  setTransactionHistory: () => null,
+  clearStatistics: () => null
+});
```

Comment:
hmm, i used it to ensure when entering the `DailyStatisticsScreen`, it will fetch the latest stats for the specific day, hence the need to clear stats. one way for this, would be to clear the stats upon pressing back (`onPressBack` in `StatisticsHeader`), would that work?

also, was taking in to account the possibility that the stats might be needed in other screens in the future, i.e. across multiple campaigns, locations etc.

Reply:
I see. In that case, how about we move the context down into `DailyStatisticsScreen` for now? In future if there's the need to share the context, we can move it up the tree. By having the context as far down the tree as possible, we avoid the need for clearing it since once `DailyStatisticsScreen` is unmounted, the context also gets unmounted.

Reply:
okay that makes sense! moved the context down to `DailyStatisticsScreen`!